From: Steve French <stfrench@microsoft.com>
Date: Thu, 15 Dec 2022 20:03:17 -0600
Subject: [PATCH] cifs: update internal module number
Git-commit: aacfc939cc42293fbcfe113040b4e8abaef68429
References: bsc#1190317
Patch-mainline: v6.2-rc1

To 2.41

Signed-off-by: Steve French <stfrench@microsoft.com>
Acked-by: Enzo Matsumiya <ematsumiya@suse.de>
---
 fs/cifs/cifsfs.h |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/fs/cifs/cifsfs.h
+++ b/fs/cifs/cifsfs.h
@@ -157,5 +157,5 @@ extern int cifs_truncate_page(struct add
 extern const struct export_operations cifs_export_ops;
 #endif /* CONFIG_CIFS_NFSD_EXPORT */
 
-#define CIFS_VERSION   "2.40"
+#define CIFS_VERSION   "2.41"
 #endif				/* _CIFSFS_H */
