From: Steve French <stfrench@microsoft.com>
Date: Wed, 19 Oct 2022 00:30:04 -0500
Subject: [PATCH] cifs: update internal module number
Git-commit: 73b1b8d25e39a1478b3792a7075f43e053ee62c2
References: bsc#1190317
Patch-mainline: v6.1-rc2

To 2.40

Signed-off-by: Steve French <stfrench@microsoft.com>
Acked-by: Enzo Matsumiya <ematsumiya@suse.de>
---
 fs/cifs/cifsfs.h |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/fs/cifs/cifsfs.h
+++ b/fs/cifs/cifsfs.h
@@ -157,5 +157,5 @@ extern int cifs_truncate_page(struct add
 extern const struct export_operations cifs_export_ops;
 #endif /* CONFIG_CIFS_NFSD_EXPORT */
 
-#define CIFS_VERSION   "2.21"
+#define CIFS_VERSION   "2.40"
 #endif				/* _CIFSFS_H */
