Subject: hv: mana: adjust mana_select_queue to old API
From: Olaf Hering <ohering@suse.de>
Patch-mainline: Never, old kernel
References: jsc#SLE-18779, bsc#1185727

--- a/drivers/net/ethernet/microsoft/mana/mana_en.c
+++ b/drivers/net/ethernet/microsoft/mana/mana_en.c
@@ -352,7 +352,7 @@ static int mana_get_tx_queue(struct net_
 }
 
 static u16 mana_select_queue(struct net_device *ndev, struct sk_buff *skb,
-			     struct net_device *sb_dev)
+			     struct net_device *sb_dev, select_queue_fallback_t fallback)
 {
 	int txq;
 
