From 774c46732ddba4632fa735beb17589aac90d5b49 Mon Sep 17 00:00:00 2001
From: "David S. Miller" <davem@davemloft.net>
Date: Wed, 16 Aug 2017 15:40:44 -0700
Subject: [PATCH] tcp: Export tcp_{sendpage,sendmsg}_locked() for ipv6.
Git-commit: 774c46732ddba4632fa735beb17589aac90d5b49
Patch-mainline: v4.14-rc1
References: bsc#1194541

Signed-off-by: David S. Miller <davem@davemloft.net>
Acked-by: João Silva <joao.silva@suse.com>
---
 net/ipv4/tcp.c |    2 ++
 1 file changed, 2 insertions(+)

--- a/net/ipv4/tcp.c
+++ b/net/ipv4/tcp.c
@@ -1053,6 +1053,7 @@ int tcp_sendpage_locked(struct sock *sk,
 
 	return do_tcp_sendpages(sk, page, offset, size, flags);
 }
+EXPORT_SYMBOL_GPL(tcp_sendpage_locked);
 
 int tcp_sendpage(struct sock *sk, struct page *page, int offset,
 		 size_t size, int flags)
@@ -1402,6 +1403,7 @@ out_err:
 	}
 	return err;
 }
+EXPORT_SYMBOL_GPL(tcp_sendmsg_locked);
 
 int tcp_sendmsg(struct sock *sk, struct msghdr *msg, size_t size)
 {
