From: Denis Kirjanov <denis.kirjanov@suse.com>
Subject: kABI workaround for net/ipvlan changes
Patch-mainline: Never, kABI workaround
References: git-fixes

Signed-off-by: Denis Kirjanov <denis.kirjanov@suse.com>

---
--- a/drivers/net/ipvlan/ipvlan.h
+++ b/drivers/net/ipvlan/ipvlan.h
@@ -74,7 +74,9 @@ struct ipvl_dev {
 	DECLARE_BITMAP(mac_filters, IPVLAN_MAC_FILTER_SIZE);
 	netdev_features_t	sfeatures;
 	u32			msg_enable;
+#ifndef __GENKSYMS__
 	spinlock_t              addrs_lock;
+#endif
 };
 
 struct ipvl_addr {
