From 7f208013797fb36ca5ac71a37d0a346679e37e7e Mon Sep 17 00:00:00 2001
From: Michal Suchanek <msuchanek@suse.de>
Date: Mon, 16 Aug 2021 22:18:01 +0200
Subject: [PATCH] kABI: s390/ap: Fix hanging ioctl caused by wrong msg counter

References: bsc#1188982 LTC#193818
Patch-mainline: Never, kABI

Signed-off-by: Michal Suchanek <msuchanek@suse.de>
---
 drivers/s390/crypto/ap_queue.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/s390/crypto/ap_queue.c b/drivers/s390/crypto/ap_queue.c
index 6f10461faf87..565aeafb0426 100644
--- a/drivers/s390/crypto/ap_queue.c
+++ b/drivers/s390/crypto/ap_queue.c
@@ -13,7 +13,9 @@
 #include <asm/facility.h>
 
 #include "ap_bus.h"
+#ifndef __GENKSYMS__
 #include "ap_debug.h"
+#endif
 
 /**
  * ap_queue_enable_interruption(): Enable interruption on an AP queue.
-- 
2.31.1

