From 8638527e85592bb3d0cffd0991dcbf0a86fded6a Mon Sep 17 00:00:00 2001
From: Michal Suchanek <msuchanek@suse.de>
Date: Thu, 26 Nov 2020 10:49:46 +0100
Subject: [PATCH] kABI: powerpc: avoid including pgtable.h in kup.h

References: CVE-2020-4788 bsc#1177666
Patch-mainline: never, kABI

Signed-off-by: Michal Suchanek <msuchanek@suse.de>
---
 arch/powerpc/include/asm/kup.h | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/arch/powerpc/include/asm/kup.h b/arch/powerpc/include/asm/kup.h
index f0f8e36ad71f..ee59808661b9 100644
--- a/arch/powerpc/include/asm/kup.h
+++ b/arch/powerpc/include/asm/kup.h
@@ -4,7 +4,9 @@
 
 #ifndef __ASSEMBLY__
 
+#ifndef __GENKSYMS__
 #include <asm/pgtable.h>
+#endif
 
 #ifdef CONFIG_PPC_BOOK3S_64
 #include <asm/book3s/64/kup-radix.h>
-- 
2.26.2

