From: Juergen Gross <jgross@suse.com>
Date: Mon, 15 Feb 2021 08:25:00 +0100
Subject: [PATCH] KVM: Fix kABI for tlb_flush
Patch-mainline: never, kABI workaround
References: bsc#1182195

Signed-off-by: Juergen Gross <jgross@suse.com>
---
 arch/x86/include/asm/kvm_host.h | 4 ++++
 1 file changed, 4 insertions(+)

--- a/arch/x86/include/asm/kvm_host.h
+++ b/arch/x86/include/asm/kvm_host.h
@@ -1013,7 +1013,11 @@ struct kvm_x86_ops {
 	void (*set_rflags)(struct kvm_vcpu *vcpu, unsigned long rflags);
 	u32 (*get_pkru)(struct kvm_vcpu *vcpu);
 
+#ifndef __GENKSYMS__
 	void (*tlb_flush)(struct kvm_vcpu *vcpu, bool invalidate_gpa);
+#else
+	void (*tlb_flush)(struct kvm_vcpu *vcpu);
+#endif
 
 	void (*run)(struct kvm_vcpu *vcpu);
 	int (*handle_exit)(struct kvm_vcpu *vcpu);
