From 42eabc218b9092bde6647f5d8606e4140cb96c46 Mon Sep 17 00:00:00 2001
From: Denis Kirjanov <denis.kirjanov@suse.com>
Date: Wed, 3 Jul 2024 13:11:12 +0300
Subject: [PATCH] net: sock: preserve kabi for sock
Patch-mainline: Never (kABI fixup)
References: bsc#1221010 CVE-2021-47103

Signed-off-by: Denis Kirjanov <denis.kirjanov@suse.com>
---
 include/net/sock.h | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/include/net/sock.h b/include/net/sock.h
index 0e4351dd0588..53cf77ec7469 100644
--- a/include/net/sock.h
+++ b/include/net/sock.h
@@ -390,7 +390,11 @@ struct sock {
 #ifdef CONFIG_XFRM
 	struct xfrm_policy __rcu *sk_policy[2];
 #endif
+#ifdef __GENKSYMS__
+	struct dst_entry        *sk_rx_dst;
+#else
 	struct dst_entry __rcu	*sk_rx_dst;
+#endif
 	struct dst_entry __rcu	*sk_dst_cache;
 	atomic_t		sk_omem_alloc;
 	int			sk_sndbuf;
-- 
2.16.4

