From 65e0a256a2c4014b2d6f284a05a10de08b46114f Mon Sep 17 00:00:00 2001
From: Denis Kirjanov <denis.kirjanov@suse.com>
Date: Mon, 8 Jul 2024 18:04:00 +0300
Subject: [PATCH 3/3] netfilter: nft_set: preserve kabi
Patch-mainline: Never, upstream has replaced the fixed code
References: bsc#1215420 CVE-2023-4244

Signed-off-by: Denis Kirjanov <denis.kirjanov@suse.com>
---
 include/net/netfilter/nf_tables.h | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/include/net/netfilter/nf_tables.h b/include/net/netfilter/nf_tables.h
index ca43e4fcd9a1..4567cd709427 100644
--- a/include/net/netfilter/nf_tables.h
+++ b/include/net/netfilter/nf_tables.h
@@ -393,8 +393,10 @@ struct nft_set {
 	u64				timeout;
 	u32				gc_int;
 	u16				policy;
+#ifndef __GENKSYMS__
 	/* alignment hole of 40 bytes due to alignemt */
 	struct kref			refs;
+#endif
 	u16				udlen;
 	unsigned char			*udata;
 	/* runtime data below here */
-- 
2.16.4

