From 4a33db745a742b398ce889529f3bca738b630924 Mon Sep 17 00:00:00 2001
From: Oliver Neukum <oneukum@suse.com>
Date: Tue, 21 Jul 2020 16:42:58 +0200
Subject: [PATCH] padata: reorder work kABI fixup
Patch-mainline: Never, kABI fixup
References: git-fixes

Signed-off-by: Oliver Neukum <oneukum@suse.com>
---
 include/linux/padata.h | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/include/linux/padata.h b/include/linux/padata.h
index 50704b5..daaa8a8 100644
--- a/include/linux/padata.h
+++ b/include/linux/padata.h
@@ -94,9 +94,11 @@ struct padata_parallel_queue {
        struct padata_list    reorder;
        struct parallel_data *pd;
        struct work_struct    work;
-       struct work_struct    reorder_work;
        atomic_t              num_obj;
        int                   cpu_index;
+#ifndef __GENKSYMS__
+       struct work_struct    reorder_work;
+#endif
 };
 
 /**
-- 
2.16.4

