From a2a462ffb01e3c7422ca47db00991053a4aec7e0 Mon Sep 17 00:00:00 2001
From: Denis Kirjanov <denis.kirjanov@suse.com>
Date: Mon, 22 Jan 2024 11:41:13 +0300
Subject: [PATCH] preserve KABI for struct sfp_socket_ops
References: git-fixes 
Patch-mainline: never, kabi

---
 drivers/net/phy/sfp.h | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/net/phy/sfp.h b/drivers/net/phy/sfp.h
index 64f54b0bbd8c..400e8f50d13b 100644
--- a/drivers/net/phy/sfp.h
+++ b/drivers/net/phy/sfp.h
@@ -7,8 +7,10 @@
 struct sfp;
 
 struct sfp_socket_ops {
+#ifndef __GENKSYMS__
 	void (*attach)(struct sfp *sfp);
 	void (*detach)(struct sfp *sfp);
+#endif
 	void (*start)(struct sfp *sfp);
 	void (*stop)(struct sfp *sfp);
 	int (*module_info)(struct sfp *sfp, struct ethtool_modinfo *modinfo);
-- 
2.16.4

