From 74f870f82f0f2022cab6f897fbd2fa1b8f886bf5 Mon Sep 17 00:00:00 2001
From: Denis Kirjanov <denis.kirjanov@suse.com>
Date: Thu, 18 Jan 2024 11:38:03 +0300
Subject: [PATCH] preserve KABI for struct plat_stmmacenet_data
References: git-fixes 
Patch-mainline: never, kabi

---
 include/linux/stmmac.h | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/include/linux/stmmac.h b/include/linux/stmmac.h
index cc2cc13cdffa..bf81c7992145 100644
--- a/include/linux/stmmac.h
+++ b/include/linux/stmmac.h
@@ -182,12 +182,14 @@ struct plat_stmmacenet_data {
 	struct clk *pclk;
 	struct clk *clk_ptp_ref;
 	unsigned int clk_ptp_rate;
-	unsigned int clk_ref_rate;
 	struct reset_control *stmmac_rst;
 	struct stmmac_axi *axi;
 	int has_gmac4;
 	bool tso_en;
 	int mac_port_sel_speed;
 	bool en_tx_lpi_clockgating;
+#ifndef __GENKSYMS__
+	unsigned int clk_ref_rate;
+#endif
 };
 #endif
-- 
2.16.4

