From: Denis Kirjanov <dkirjanov@suse.de>
Subject: qed: fix kABI in qed_rdma_create_qp_in_params
Patch-mainline: Never, kABI fix
References: git-fixes

Signed-off-by: Denis Kirjanov <dkirjanov@suse.de>

---
 include/linux/qed/qed_rdma_if.h |    2 ++
 1 file changed, 2 insertions(+)

--- a/include/linux/qed/qed_rdma_if.h
+++ b/include/linux/qed/qed_rdma_if.h
@@ -320,7 +320,9 @@ struct qed_rdma_create_qp_in_params {
 	u64 rq_pbl_ptr;
 	u16 srq_id;
 	u8 stats_queue;
+#ifndef __GENKSYMS__
 	u8 flags;
+#endif
 #define QED_ROCE_EDPM_MODE_MASK      0x1
 #define QED_ROCE_EDPM_MODE_SHIFT     0
 };
