From: Petr Mladek <pmladek@suse.com>
Subject: kABI workaround for usermodehelper changes
Patch-mainline: Never, kABI workaround for SLE15 & co
References: bsc#1179406

Signed-off-by: Petr Mladek <pmladek@suse.com>
---
 kernel/kmod.c |    2 ++
 1 file changed, 2 insertions(+)

--- a/kernel/kmod.c
+++ b/kernel/kmod.c
@@ -30,7 +30,9 @@
 #include <linux/cred.h>
 #include <linux/file.h>
 #include <linux/fdtable.h>
+#ifndef __GENKSYMS__
 #include <linux/fs_struct.h>
+#endif
 #include <linux/workqueue.h>
 #include <linux/security.h>
 #include <linux/mount.h>
